hypothesis-python/tests/numpy/test_gen_data.py

```diff
@@ -340,6 +340,12 @@ def test_array_values_are_unique(arr):
     assert len(set(arr)) == len(arr)
 
 
+def test_cannot_generate_unique_array_of_too_many_elements():
```

Comment:
What's this test doing in this PR?

Reply:
Coverage for the non-unique-element-rejected branch was flaky without it!

Reply:
> I can pull it out to a separate PR if you like, it's just here because the build failed for this PR first.

Which demonstrates the problem with doing it this way! This PR has been delayed a while, while a PR with this patch would not have been, and as a result we've had a bunch of flaky builds in the interim.

Reply:
I haven't seen it flake anywhere else, but it's split out as #2232 now.  

(I do tend to batch things up a little for review, but only to the point that it's faster in expectation)